spec.bs

```diff
@@ -232,7 +232,122 @@ to conduct an auction to select an advertisement to display to the user, they ca
 {{Window/navigator}}.{{Navigator/runAdAuction()}} function, providing an auction configuration that
 tells the browser how to conduct the auction and which on-device recorded interests are allowed to
 bid in the auction for the chance to display their advertisement.
-  
+
+<h3 id="script-runners">Script Runners</h3>
+
+<xmp class="idl">
+[SecureContext]
+interface InterestGroupBiddingScriptRunner {
+  [NewObject] Promise<undefined> createScriptRunner(USVString script);
+};
+
+[SecureContext]
+interface InterestGroupScoringScriptRunner {
+  [NewObject] Promise<undefined> createScriptRunner(USVString script);
+};
+
+[SecureContext]
+interface InterestGroupReportingScriptRunner {
+  [NewObject] Promise<undefined> createScriptRunner(USVString script);
+};
+
+[SecureContext]
+interface InterestGroupScriptRunner {
+};
+</xmp>
+
+Note: WebIDL interfaces exposed to `*` are not exposed to interest group script runners.
+
+<xmp class="idl">
+[Exposed=InterestGroupBiddingScriptRunner,
+Global=(InterestGroupScriptRunner, InterestGroupBiddingScriptRunner)]
+interface InterestGroupBiddingScriptRunnerGlobalScope{
+  boolean setBid(optional GenerateBidOutput bid);
+  undefined setPriority(double priority);
+  undefined setPrioritySignalsOverride(DOMString key, double priority);
+};
+
+[Exposed=InterestGroupScoringScriptRunner,
+Global=(InterestGroupScriptRunner, InterestGroupScoringScriptRunner)]
+interface InterestGroupScoringScriptRunnerGlobalScope{
+};
+
+[Exposed=InterestGroupReportingScriptRunner,
+Global=(InterestGroupScriptRunner, InterestGroupReportingScriptRunner)]
+interface InterestGroupReportingScriptRunnerGlobalScope{
+  attribute DOMString reportWinURL;
+  attribute DOMString reportSellerURL;

```

Comment:
Do we need these DOMStrings?  They aren't really exposed, right?

Reply:
removed.